src/App.tsx

```diff
@@ -1,61 +1,135 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>(todos);
+
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+
+  const [option, setOption] = useState(0);
+  const [userError, setUserError] = useState(false);
+
+  const pushUser = (todo: Todo) => {
+    const newTodo = {
+      ...todo,
+    };
+
+    setCurrentTodos(currentTodo => [...currentTodo, newTodo]);
+  };
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    setTitle(value);
+    setTitleError(false);
+  };
+
+  const handleUserChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = e.target;
+
+    setOption(+value);
+    setUserError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setOption(0);
+    setTitleError(false);
+    setUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (option === 0) {
+      setUserError(true);
+    }
+
+    if (title && option !== 0) {
+      pushUser({
+        id: Math.max(...todos.map(({ id }) => id)) + 1,
+        title,
+        completed: false,
+        userId: 0,
+        user: getUser(option),
+      });
+
+      reset();
+    }
+  };
```

Comment:
Can you modify this function a bit, so this part of the code is not inside of `if` statement?

What I mean is, you are already checking title and option at the beginning of the , I would like see how would you approach this ;)
```
      pushUser({
        id: Math.max(...todos.map(({ id }) => id)) + 1,
        title,
        completed: false,
        userId: 0,
        user: getUser(option),
      });

      reset();
```